tests/tex/formats.cpp

```diff
@@ -48,38 +47,91 @@ constexpr auto k_best_formats = BestFormatFor{.uncompressed               = DXGI
                                               .compressed                 = DXGI_FORMAT_BC7_UNORM,
                                               .compressed_without_alpha   = DXGI_FORMAT_BC5_UNORM};
 
+using btu::tex::OpaqueAlpha, btu::tex::AllowCompressed;
+
 TEST_CASE("guess_best_format compressed with alpha", "[src]")
 {
-    auto result = guess_best_format(DXGI_FORMAT_BC3_UNORM, k_best_formats, Yes);
+    auto result = guess_best_format(DXGI_FORMAT_BC3_UNORM, k_best_formats, OpaqueAlpha::No, AllowCompressed::Yes);
     CHECK(result == DXGI_FORMAT_BC7_UNORM);
 }
 
+TEST_CASE("guess_best_format compressed with opaque alpha", "[src]")
+{
+    auto result = guess_best_format(DXGI_FORMAT_BC3_UNORM,
+                                    k_best_formats,
+                                    OpaqueAlpha::Yes,
+                                    AllowCompressed::Yes);
+    CHECK(result == DXGI_FORMAT_BC5_UNORM);

```

Comment:
Please use `k_best_formats.XXX` instead of the value. The rest of the tests should be eventually refactored as well

Reply:
Refactored all the tests for this function.